src/ashell/file-utils.c

```diff
@@ -39,6 +39,12 @@ int fs_exist(const char *path)
 
 fs_file_t *fs_open_alloc(const char *filename, const char *mode)
 {
+#ifndef ZJS_ASHELL
+    if (mode[0] == 'w') {
+        DBG_PRINT("Write not enabled on FS for non ASHELL builds\n");
+        return NULL;
+    }
+#endif // !ZJS_ASHELL
```

Comment:
This was to make the FS read only for non ASHELL builds.  The idea being once you built, the device wouldn't be able to save and run malicious JS.

Reply:
Yeah, but again ashell is is 'insecure mode' so I don't think we need to worry about that. They also wouldn't be able to save and run non-malicious JS, which could be cool. :)

Reply:
Ah ok good point, if the only way to write stuff to the filesystem is by using ASHELL - we're already 'insecure' because who knows what they wrote to the FS.

Reply:
That's true, I think you made my point better than I did. :)